parser.go

```diff
@@ -1207,8 +1207,14 @@ func (parser *Parser) parseField(pkgName string, field *ast.Field) (*structField
 	}
 
 	jsonTag := structTag.Get("json")
+	hasStringTag := false
 	// json:"tag,hoge"
 	if strings.Contains(jsonTag, ",") {
+		// json:"name,string" or json:",string"
+		if strings.Contains(jsonTag, ",string") {
+			hasStringTag = true
```

Comment:
I think one line here is enough
```
structField.schemaType = "string"
```

Reply:
Hi @sdghchj .

Indeed, that's the fix. (line 1364)
But in order to have working JSON examples, the other stuff is needed. 
Let me clarify this change:

1. identify `json=",string"` tag [1213-1217]
1. if the example tag exists, do not convert from the string, just keep it [1258-1261]
1. in the end (so, I will keep processing tags like min, max, etc... which depend on the original type) 
1.1 I fix the bug [1364]
1.1 If no example is manually set before, I force an example (otherwise, we would have "string" as an example for an int, which would not marshal as well)

Let me know if you need me to clarify anything else. 
I tried to keep these clarifications in the code, but I'm the only one inside my head, so help me to make it clearer

Reply:
In my opinion, ",string" is like tag `swaggertype:"string"`, so one himself has to make sure the original type can be converted to string directly or by implementing some interface, and know clearly how to define examples,default values, etc. , however, swag can't guarantee it. 
As for tags min,max etc. , they base on the new type specified by tag swaggertype, so it may be a little confusing if they bases on the original type in the case of ",string",  I think.

As you known,",string" only affects integer,string,number,boolean,  if the original type is something else, lines [1258-1261] and [1366-1371] will drop examples. 
 
I prefer to use defer to do some work at the end of a function if strong related codes cross too long to the end. That will looks graceful.

I am not sure whether it is suitable to force an example only in the case of tag ",string".

Reply:
I understand you, but let me give you just one last argument. 
If you disagree, let me know and I will simplify my PR with your input.

My goal was to make sure you can use the swagger-ui with minimum stress. 
So, in my opinion, when you "try it out" with the defaults, you should get a default behaviour as well. It should "just work".

The default behaviour with `json=",string"` is (un)marshaling as it was the original type.
So, when you "try it out", swagger-ui will force an example on you with invalid values unless you force it somehow. 
I don't agree with this being like tag `swaggertype:"string"` because then you are already focusing on swagger (you will probably set a working example anyway), which is not the case with the `,string` tag.

Maybe the `parser.go` is the wrong place for this functionality (point me to the right place then), but if you don't set some example, an `int` will always fail to unmarshal with a "string".

Reply:
OK then